Runtime/Core/ApiGatewayManagement/AmazonGameLiftClientWrapper.cs

```diff
@@ -42,17 +57,191 @@ public async Task<SearchGameSessionsResponse> SearchGameSessions(SearchGameSessi
             return await _amazonGameLiftClient.SearchGameSessionsAsync(request);
         }
 
-        public static IAmazonGameLift Create(string localEndpoint)
+        public async Task<DescribeGameSessionsResponse> DescribeGameSessions(DescribeGameSessionsRequest request)
+        {
+            return await _amazonGameLiftClient.DescribeGameSessionsAsync(request);
+        }
+
+        private IAmazonGameLift Create()
+        {
+            var credentials = SetupCredentials();
+            return new AmazonGameLiftClient(credentials.AccessKey, credentials.SecretKey);
+        }
+        #endregion
+        
+        /// <summary>
+        /// Server region is code dedicated to Amazon GameLift SDK and AWS SDK calls made by the game server. All of these calls will be done via UI Elements or on Startup. 
+        /// </summary>
+        #region Server
+        
+        public AmazonGameLiftClientWrapper()
+        {
+            _gameLiftClientSettings = Resources.FindObjectsOfTypeAll<GameLiftClientSettings>()[0];
+            _amazonGameLiftClient = Create();
+        }
+
+        private async Task<ListLocationsResponse> ListLocations(ListLocationsRequest request)
+        {
+            return await _amazonGameLiftClient.ListLocationsAsync(request);
+        }
+
+        private async Task<CreateLocationResponse> CreateLocation(CreateLocationRequest request)
+        {
+            return await _amazonGameLiftClient.CreateLocationAsync(request);
+        }
+
+        private async Task<RegisterComputeResponse> RegisterCompute(RegisterComputeRequest request)
+        {
+            return await _amazonGameLiftClient.RegisterComputeAsync(request);
+        }
+
+        private async Task<GetComputeAuthTokenResponse> GetComputeAuthToken(GetComputeAuthTokenRequest request)
+        {
+            return await _amazonGameLiftClient.GetComputeAuthTokenAsync(request);
+        }
+
+        private Task<CreateFleetResponse> CreateFleet(CreateFleetRequest request)
+        {
+            return _amazonGameLiftClient.CreateFleetAsync(request);
+        }
+        
+        private RetriveAwsCredentialsResponse RetrieveAwsCredentials(string profileName)
+        {
+            var request = new RetriveAwsCredentialsRequest { ProfileName = profileName };
+            return _credentialsStore.RetriveAwsCredentials(request);
+        }
+        
+        public async Task UpdateAuthToken()
+        {
+            try
+            {
+                await Task.WhenAll(CreateFleet(), RegisterCompute(), GenerateAuthToken());
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
+        }
```

Comment:
Is this just meant to be a helper to perform the 3 actions together? If so, please rename the method to be representative of the 3 tasks being performed. 

Is this being used anywhere? Can't seem to find references - if it's not being used, please consider cleaning up this code.